src/controllers/quest.js

```diff
@@ -386,175 +382,21 @@ class Quest extends Controller {
-						} else {
-							str = str.replace('Evolve', 'Use a item to evolve')
-						}
-						break
-					case 12:
-						str = str.replace('pokéstop(s)', "pokéstop(s) you haven't visited before")
-						break
-					case 14:
-						str = str.replace('Land', 'Make')
-						if (typeof questinfo.throw_type_id === 'undefined') {
-							str = str.replace('throw(s)', 'Throw(s) in a row')
-						} else {
-							str = str.replace('throw(s)', `${this.GameData.utilData.throwType[questinfo.throw_type_id]} Throw(s) in a row`)
-						}
-						if (item.conditions[1] && item.conditions[1].type === 15) {
-							str = str.replace('Throw(s)', 'Curveball Throw(s)')
-						}
-						break
-					case 22:
-						str = str.replace('Win', 'Battle a Team Leader').replace('pvp battle(s)', 'times')
-						break
-					case 23:
-						str = str.replace('Win', 'Battle Another Trainer').replace('pvp battle(s)', 'times')
-						break
-					case 25:
-						str = str.replace('{0} pokémon', `pokémon caught ${questinfo.distance}km apart`)
-						break
-					case 27:
-						if (questinfo.character_category_ids.length > 1) {
-							let first = true
-							for (const [index, charId] of Object.entries(questinfo.character_category_ids)) {
-								if (first) {
-									gstr += `${gruntCharacterTypes[charId]}`
-								} else {
-									gstr += (index === questinfo.character_category_ids.length - 1) ? ` or ${gruntCharacterTypes[charId]}` : `, ${gruntCharacterTypes[charId]}`
-								}
-								first = false
-							}
-						} else {
-							gstr += `${gruntCharacterTypes[questinfo.character_category_ids]}`
-						}
-						str = str.replace('Team GO Rocket Grunt(s)', gstr)
-						if (item.conditions[1] && item.conditions[1].type === 18) {
-							str = str.replace('Battle against', 'Defeat')
-						}
-						break
-					case 28:
-						if (item.quest_type === 28) {
-							str = str.replace('Snapshot(s)', 'Snapshot(s) of your Buddy')
-						}
-						break
-					default:
-				}
-			} else if (item.type > 0) {
-				switch (item.type) {
-					case 7:
-						str = str.replace('Complete', 'Battle in a gym').replace('gym battle(s)', 'times')
-						break
-					case 8:
-						str = str.replace('Complete', 'Battle in a raid').replace('raid battle(s)', 'times')
-						break
-					case 13:
-						str = str.replace('Catch', 'Use').replace('pokémon with berrie(s)', 'berries to help catch Pokémon')
-						break
-					case 17:
-						str = str.replace('Walk your buddy to earn', 'Earn').replace('candy', 'candy walking with your buddy')
-						break
-					default:
-				}
-			}
-			str = str.replace('{0}', item.target)
-			if (item.target === 1) {
-				str = str.replace('(s)', '').replace('1 ', 'a ').replace(' a times', '').replace('friends', 'friend')
-			} else {
-				str = str.replace('(s)', 's')
+		if (item.title) {
+			item.quest_title = item.title
+		}
+		const questinfo = `quest_title_${item.title}`
+		const questTitle = translations[this.config.general.locale].questTitles
```

Comment:
This is translating once? Surely this should be translating based on the human's current settings?
Should also be a default for the unknownQuestString if one is not configured
GameData should also be in this.GameData so suspect translations should pop through as this.GameData.translations and not need importing directly